### ISSUE TITLE
Horizontal scroll event

### ISSUE THREAD

**Issue by viluon** [dan200/ComputerCraft#569]
It'd be nice if there were a `horizontal_scroll` (/`horiz_scroll`/`hor_scroll`) event, or an extra parameter to the existing `scroll` event, either a boolean indicating the scroll axis, or another number equal to the amount of horizontal scrolling performed.

This would enable programs to react on full 2D scrolling on notebook touchpads, for example.

**Comment by nothjarnan** [comment]
If you're talking about [mouse_scroll](http://computercraft.info/wiki/Mouse_scroll_(event)), it could probably be as simple as to have `os.pullEvent("mouse_scroll")`
return another parameter for horizontal scroll direction, like it does with vertical.
so the output would be `event, verticalDirection, horizontalDirection, x,y`.

**Comment by SquidDev** [comment]
I feel it'd be better to do this as a separate event. There's probably a fair few programs that will not cope with the scroll delta being zero (doing `if delta > 0 then else` and what not).

**Comment by nothjarnan** [comment]
fair point. so a `mouse_scroll_horizontal` event would be doable?

**Comment by nothjarnan** [comment]
@hugeblank Allow for more mouse buttons? The one issue I could see with that is that programs are made that only work with certain mice, or a mouse that has x amount of buttons. But then it'd just be as easy as asking the author to make the buttons re-bindable, to be fair.

**Comment by viluon** [comment]
@nothjarnan @hugeblank while this isn't a conversation well suited for this issue, strictly speaking, I believe ComputerCraft already supports more buttons — any extra buttons are simply assigned higher numbers (for example, my back button triggers an event, and is listed as button `5`). This may be a perk of [CCEmuX](https://emux.cc), however.

**Comment by hugeblank** [comment]
I think so. I have tried to use buttons in game and iirc this doesn't work. May have been a CC:T thing too.

**Comment by SquidDev** [comment]
Yeah, CC explicitly guards against that. CCEmuX really should too:

https://github.com/dan200/ComputerCraft/blob/914df8b0c7cc28451ea56a53808024511ff52642/src/main/java/dan200/computercraft/client/gui/widgets/WidgetTerminal.java#L182

**Comment by AnCarsenat** [comment]
horizontal mouse scroll doesnt exist
